packages/rs-drive-abci/src/execution/platform_events/voting/check_for_ended_vote_polls/v0/mod.rs

```diff
@@ -66,31 +67,26 @@ where
                             .sorted_by(|a, b| Ord::cmp(&b.final_vote_tally, &a.final_vote_tally))
                             .collect();
 
-                        let restrict_to_only_fetch_contenders = if sorted_contenders
-                            .last()
-                            .map(|last| last.final_vote_tally > 0)
-                            .unwrap_or_default()
+                        let (contenders_with_votes, contenders_with_no_votes) : (Vec<_>, Vec<_>) = sorted_contenders.iter().partition(|a| a.final_vote_tally > 0);
+
+                        let (restrict_to_only_fetch_contenders, mut other_contenders) = if contenders_with_no_votes.is_empty()
                         {
-                            None
+                            (None, contenders_with_no_votes.into_iter().map(|contender| (TowardsIdentity(contender.identity_id), vec![])).collect())
```

Comment:
if contenders_with_no_votes.is_empty() then we return (None,None) because `contenders_with_no_votes.into_iter()` create empty iterator.

Reply:
fixed